lib/logmsg/logmsg.c

```diff
@@ -968,30 +998,24 @@ log_msg_append_format_sdata(const LogMessage *self, GString *result,  guint32 se
       sdata_elem = sdata_name + 7;
       sd_id_len = (handle_flags >> 8);
 
-      if (sd_id_len)
+      dot = find_sd_id_in_key(sd_id_len, sdata_elem, sdata_name, sdata_name_len);
```

Comment:
I just checked, and nv_registry handle_flags would _NEVER_ contain the length of the SD ID, so sd_id_len is _always_ zero.

As it seems I've removed its setting in e85b77daf7adefc20186784f538ac642f414fb6b. Originally the intention has been to store the length of the SD ID, so that our joining of the name-value pair wouldn't cause a change in the SDATA format.

I am not sure if the removal was the right move, but definitely, we would also need to remove the code conditional on sd_id_len != 0. 

That trick might be used to "cache" our decision on the separation of the SD-ID and the SD-PARAM, so this code could be faster. But instead of doing this in syslogformat (where I've removed from), it should be added to the SDATA slow path, e.g. the point where an SDATA nvpair is registered and get applied to any kind of SDATA nvpair, be them parsed from an actual RFC5424 message, or set in a different place (E.g. json-parser()).